src/store/farms/actions.js

```diff
@@ -1818,10 +1867,11 @@ export default {
   },
 
   async harvestAllFarms({ commit, state, dispatch }) {
-    const farmContract = await getContract(state.contract);
     const batch = getBatch();
     const harvestingFarms = [];
     for (const farmId in state.data) {
+      const contract = getFarmContract(farmId, state);
+      const farmContract = await getContract(contract);
```

Comment:
This is going to load the contract for every farm you are in, but there are a max of 2 contracts. Loading a contract into taquito is expensive. If someone is in 10 farms, we don't need to load the contract 10 times. There was a previous optimization here to load it once which is now gone. We only need to load it a max of twice. Simplest fix is to just load them both before the loop and use the correct one as needed.